### ISSUE TITLE
warning: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int' [-Wshorten-64-to-32]

### ISSUE THREAD

**Issue by yurivict** [glotzerlab/hoomd-blue#1205]
```
/disk-samsung/freebsd-ports/science/py-hoomd-blue/work/hoomd-blue-3.0.0-beta.13/hoomd/MemoryTraceback.cc:35:23: warning: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int' [-Wshorten-64-to-32]
    int num_symbols = backtrace(&m_traces[idx].front(), MAX_TRACEBACK);
        ~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

```backtrace``` is defined this way:
```
     #include <execinfo.h>

     size_t
     backtrace(void **addrlist, size_t len);
```

Another problem with ```backtrace``` is that it also requires the library ```execinfo``` to be linked to on FreeBSD.

Version: 3.0.0-beta.13
clang-12
FreeBSD 13

**Comment by joaander** [comment]
HOOMD-blue is open source. Feel free to submit a pull request to fix this.

**Comment by yurivict** [comment]
Fix for the warning: https://github.com/glotzerlab/hoomd-blue/pull/1209

**Comment by yurivict** [comment]
Second commit in the same PR fixes the ```execinfo``` link problem on FreeBSD.

**Comment by joaander** [comment]
Fixed in #1209
